lib/puppet/provider/vcsrepo/git.rb

```diff
@@ -91,6 +92,20 @@ def revision=(desired)
     update_owner_and_excludes
   end
 
+  def includes
+    return nil if bare_exists?
+
+    at_path do
+      return nil unless File.file?(File.join(@resource.value(:path), '.git/info/sparse-checkout'))
+      File.readlines(File.join(@resource.value(:path), '.git/info/sparse-checkout')).map(&:chomp)
+    end
```

Comment:
`at_path` is redundant if you use `@resource.value(:path)`.  You can either skip it completely:

```suggestion
    return nil unless File.file?(File.join(@resource.value(:path), '.git/info/sparse-checkout'))
    File.readlines(File.join(@resource.value(:path), '.git/info/sparse-checkout')).map(&:chomp)
```

or (feels simpler) keep it and use relative paths:

```suggestion
    at_path do
      return nil unless File.file?('.git/info/sparse-checkout')
      File.readlines('.git/info/sparse-checkout').map(&:chomp)
    end
```